### ISSUE TITLE
SDL dinput broke Wiimote support

### ISSUE THREAD

**Issue by mat8913** [ValveSoftware/wine#19]
In Wine 3.7, I was able to use my Wii Classic Controller to play Hyperdimension Neptunia. In Proton's Wine, the game no longer detects it. However, it still shows up in `wine control` under "Game Controllers" as "Nintendo Wii Remote Classic Controller (js)".

This patch applied to Proton's Wine fixes the issue:

```patch
diff --git a/dlls/dinput/dinput_main.c b/dlls/dinput/dinput_main.c
index 91119548d5..0fc384be1c 100644
--- a/dlls/dinput/dinput_main.c
+++ b/dlls/dinput/dinput_main.c
@@ -88,9 +88,9 @@ static const struct dinput_device *dinput_devices[] =
 {
     &mouse_device,
     &keyboard_device,
-    &joystick_sdl_device,
     &joystick_linuxinput_device,
     &joystick_linux_device,
+    &joystick_sdl_device,
     &joystick_osx_device
 };
 #define NB_DINPUT_DEVICES (sizeof(dinput_devices)/sizeof(dinput_devices[0]))
```

I'm super unfamiliar with the codebase so this might be a bit of a stupid question, but why stop enumeration after SDL?

**Comment by aeikum** [comment]
We don't want devices to appear multiple times, once through SDL and again through the js or event interfaces. The right way to fix this would be to see why SDL isn't enumerating the device as a joystick.

**Comment by mat8913** [comment]
SDL only enumerates devices with the `ID_INPUT_JOYSTICK` property, which the hid-wiimote driver does not set. The hid-wiimote developers don't plan on changing this (related issue: dvdhrm/xwiimote#51).

For now I'm using xboxdrv to emulate an XBox controller as a workaround.

**Comment by mat8913** [comment]
I upgraded to Debian Buster and this is now fixed. I'm not entirely sure, but I think it was fixed by systemd/systemd@2e856c63d466455f91bed3046265beffb5235044.
